lib/core/src/animation/fixed_step_animator.rs

```diff
@@ -0,0 +1,48 @@
+use super::DifferentialTimeAnimator;
+use super::FnAnimation;
+
+use nalgebra::zero;
+
+// =============================
+// === FixedStepAnimatorData ===
+// =============================
+
+struct FixedStepAnimatorData {
+    closure          : Box<dyn FnMut(f32)>,
+    step_duration    : f32,
+    accumulated_time : f32
+}
+
+impl FixedStepAnimatorData {
+    pub fn new<F:FnAnimation>(steps_per_second:f32, f:F) -> Self {
+        let closure          = Box::new(f);
+        let step_duration    = 1000.0 / steps_per_second;
+        let accumulated_time = zero();
+        Self { closure,step_duration,accumulated_time }
+    }
+}
+
+
+
+// ================
+// === Animator ===
+// ================
+
+/// This structure is aimed to run a closure at a fixed time rate.
```

Comment:
But this sentence is not true. Frames can take longer than 1/step_duration. So you cannot guarantee that this structure runs a closure at fixed time rate. This sentence is misleading.

Reply:
Welp, you can aim a target, but it can miss it :D

```rust
/// This structure attempts to run a closure at a fixed time rate.
```

Sounds better?

Reply:
Much better. But the docs should explain more. What happens if it fails - that it runs the closure couple of times etc. I know it, you know it, but the reader of the docs should know it as well :D

Reply:
@wdanilo this?
```rust
/// This structure attempts to run a closure at a fixed time rate.
///
/// # Internals
/// If, for instance, we want to run FnAnimation once per second, it's step duration
/// (FnAnimation(delta_time)) will be 1 second. But keep in mind that if the actual frame takes
/// longer, say 2 seconds, FnAnimation will be called twice in the same moment, but its delta_time
/// parameter will always be fixed to 1 second.
```